### PR TITLE
Add `pluginGithubReleaseTagPattern` option to plugin config schema

### PR DESCRIPTION
**Pull request by panusoi**
Added `pluginGithubReleaseTagPattern` option to plugin config schema.  

Allows the developer of the plugin to configure where release notes are fetched, useful e.g. in monorepos where the latest release tag may be related to a different package. The `pluginGithubReleaseTagPattern` should be a string and can include `%VERSION%` which is replaced with the latest version number. E.g. the latest version is 0.2.0 and the pattern is "`homebridge-example-plugin@%VERSION%`", in which case the release notes are fetched from release with a tag `homebridge-example-plugin@0.2.0`. If `pluginGithubReleaseTagPattern` is not set , the release notes are fetched as before, i.e. from the latest tag

### CODE REVIEW

**Comment by bwp91** [comment]
Not sure how I feel about this. I understand why you have suggested these changes. But I don't think that the plugin config schema is the place to start adding options, not sure I would want to start a precedent here. any thoughts @NorthernMan54 ?

I believe homebridge-ring has a monorepo, @dgreif how do you deal with release notes in the homebridge ui with a monorepo? Can you see any benefit to this PR?

**Comment by dgreif** [comment]
I use [Changesets](https://github.com/changesets/changesets) for the release notes in `dgreif/ring`. It's got a nice CLI for managing release notes on a per-package basis in a monorepo. The only downside is the GitHub Releases are somewhat limited in what you can do in a monorepo, so you only get one "Release" per version number

**Comment by bwp91** [comment]
thanks for your input here @dgreif !
i am not against the idea of the changes of this PR, i am just hesitant of setting a precedent that options can be added willynilly to the config schema file

File: README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ For detailed instructions on how to setup Node.js and Homebridge with Homebridge
 * <img src="https://user-images.githubusercontent.com/3979615/59594527-56fa8c00-9137-11e9-937b-32092dfcff41.png" alt="docker" height="16px" width="16px"/> [Setup Homebridge using Docker](https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Docker)
 * <img src="https://user-images.githubusercontent.com/3979615/78118531-dc46f700-7452-11ea-95e5-977f79d1904f.png" alt="synology-dsm" height="16px" width="16px"/> [Setup Homebridge on a Synology NAS](https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Synology-DSM)
 
-If your platform is not listed above, or you want to use your own service manager, see the [Manual Configuration](https://github.com/oznu/homebridge-config-ui-x/wiki/Manual-Configuration) wiki article for instructions on setting up the Homebridge UI to run as a Homebridge plugin instead of a service.
+If your platform is not listed above, or you want to use your own service manager, see the [Manual Configuration](https://github.com/homebridge/homebridge-config-ui-x/wiki/Manual-Configuration) wiki article for instructions on setting up the Homebridge UI to run as a Homebridge plugin instead of a service.
```
Comment (bwp91) [inline README.md:38]: these wiki changes would be great in a separate PR which I would be happy to merge immediately 😀

File: README.md [hunk 2]
```diff
@@ -128,11 +128,11 @@ Make sure you are running [supported versions of node and npm](#supported-nodejs
 
 #### 2. Accessories tab missing
 
-If the Accessories tab is not shown then you are not running Homebridge in insecure mode. See the [Enabling Accessory Control](https://github.com/oznu/homebridge-config-ui-x/wiki/Enabling-Accessory-Control) wiki for details. If you have just enabled insecure mode make sure you have restarted Homebridge and refreshed the page in your browser.
+If the Accessories tab is not shown then you are not running Homebridge in insecure mode. See the [Enabling Accessory Control](https://github.com/homebridge/homebridge-config-ui-x/wiki/Enabling-Accessory-Control) wiki for details. If you have just enabled insecure mode make sure you have restarted Homebridge and refreshed the page in your browser.
```
Comment (bwp91) [inline README.md:131]: see above wiki change comment

File: README.md [hunk 3]
```diff
@@ -128,11 +128,11 @@ Make sure you are running [supported versions of node and npm](#supported-nodejs
...
 
 #### 3. Running in Docker
 
-The Homebridge UI supports the [oznu/homebridge](https://github.com/oznu/docker-homebridge) Docker image. You must enable the UI using the method described in [the wiki](https://github.com/oznu/homebridge-config-ui-x/wiki/Enabling-UI-with-Docker).
+The Homebridge UI supports the [homebridge/homebridge](https://github.com/homebridge/docker-homebridge) Docker image. You must enable the UI using the method described in [the wiki](https://github.com/homebridge/homebridge-config-ui-x/wiki/Enabling-UI-with-Docker).
```
Comment (bwp91) [inline README.md:135]: see above wiki change comment

File: src/modules/plugins/plugins.service.ts [hunk 4]
```diff
@@ -859,8 +859,11 @@ export class PluginsService {
       throw new NotFoundException();
     }
 
+    const releaseTagPattern = plugin.settingsSchema ? await this.getPluginGithubReleaseTagPattern(pluginName) : null;
```
Comment (bwp91) [inline src/modules/plugins/plugins.service.ts:862]: does this need to default to `null` anyway if the function itself returns null in 'else' cases?

**Comment by bwp91** [comment]
I haven't checked this properly with regards to this PR, but at least in this file there are some hardcoded references to the `latest` branch.

https://github.com/homebridge/homebridge-config-ui-x/blob/master/ui/src/app/core/manage-plugins/manage-plugins-modal/manage-plugins-modal.component.ts

I wonder if anything here would need to be updated as part of this PR?
